api/v3/OdooSync.php

```diff
@@ -9,8 +9,22 @@
  * @throws \Exception
  */
 function civicrm_api3_odoo_sync_run($params) {
-  $sync = new CRM_Odoosync_Sync_Contact($params);
-  return $sync->run();
+  $log = [];
+  $logContact = (new CRM_Odoosync_Sync_Contact($params))->run();
```

Comment:
the content of this API method should be abstracted into seperate class, something like CRM_Odoosync_Sync::Run()